cpp/src/arrow/acero/task_util.cc

```diff
@@ -413,6 +421,8 @@ void TaskSchedulerImpl::Abort(AbortContinuationImpl impl) {
             all_finished = false;
             task_group.state_ = TaskGroupState::ALL_TASKS_STARTED;
           }
+        } else if (task_group.state_ == TaskGroupState::ALL_TASKS_STARTED) {
```

Comment:
This is needed for allowing the `Abort` to be called in a task itself, otherwise the abort continuation will be called twice.

I suppose the `Abort` function is not necessarily to be called inside a task body to trigger the issue. In other words, it could happen when `Abort` is called in a timing that is subtle enough.

Reply:
Minor nit: can we change this to a switch?

Reply:
OK, done.